src/idpi/product.py

```diff
@@ -43,3 +44,45 @@ def __call__(self, *args, **kwargs):
     @property
     def descriptor(self):
         return self._desc
+
+
+def _merge_requests(
+    products: list[Product],
+) -> tuple[dict[str, Request], dict[str, list[str]]]:
+    req_labels: dict[Request, set[str]] = {}
+    for product in products:
+        for name, req in product.descriptor.input_fields.items():
+            name_set = req_labels.setdefault(req, set())
+            name_set.add(name)
+
+    # check for any conflicts in the names
+    cnt = Counter(name for names in req_labels.values() for name in names)
+    conflicts = [name for name, count in cnt.items() if count > 1]
+    if conflicts:
+        raise RuntimeError(f"These names are involved in conflicts: {conflicts}")
```

Comment:
Its not immediately clear what the context of this conflict check is for, can you add some more info to the comment? Also is name representing param here?

Reply:
We are merging mappings of names to requests, it's okay if there are multiple names for a single request but not if there are multiple requests for a single name. The name is indeed used as a key in the output dataset. The name references a field request and the corresponding `xarray.DataArray` returned by the data source.